aten/src/THCS/generic/THCSTensor.cpp

```diff
@@ -120,6 +126,29 @@ THCSTensor* THCSTensor_(_set)(THCState *state, THCSTensor *self, THCIndexTensor
   return THCSTensor_(_move)(state, self, THCIndexTensor_(newClone)(state, indices), THCTensor_(newClone)(state, values));
 }
 
+THCSTensor* THCSTensor_(_move_csr)(THCState *state, THCSTensor *self, THCudaIntTensor *csr) {
+  int empty = (csr == NULL);
+  if (!empty) {
+    THArgCheck(self->size[0] + 1 == THCudaIntTensor_size(state, csr, 0), 1, 
+        "csr must be of length of first dimension + 1, expected %d, got %d", self->size[0] + 1,
+        THCudaIntTensor_size(state, csr, 0));
+  }
+  if (self->csr != NULL) {
+    THCudaIntTensor_free(state, self->csr);
+  }
+  self->csr = csr;
+
+  return self;
+}
+
+THCSTensor* THCSTensor_(_set_csr)(THCState *state, THCSTensor *self, THCudaIntTensor * csr) {
+  if (csr != NULL) {
+    return THCSTensor_(_move_csr)(state, self, THCudaIntTensor_newClone(state, csr));
```

Comment:
I don't think we should clone the CSR. It's a lazily computed attribute, and it might not be needed by the clone